server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/MaxValueAggregator.java

```diff
@@ -65,11 +65,16 @@ public Long toLongValue(Double value) {
     }
 
     @Override
-    public Double toStarTreeNumericTypeValue(Long value, StarTreeNumericType type) {
+    public Double toStarTreeNumericTypeValue(Long value) {
         try {
-            return type.getDoubleValue(value);
+            return VALUE_AGGREGATOR_TYPE.getDoubleValue(value);
```

Comment:
This should be based on the index mapping field type rather than the value aggregator type. ( i know i recommended this initially :P  but unit tests were failing and i had to change )

I have changed the implementation as part of merge PR lets check it out.

The real concern is why tests didn't catch this - lets identify why unit test didn't catch this and add some unit tests.

Applies to all aggregators.

Reply:
I think this was pulled in from your commit in merge PR. When I rebase it totally, it should be fixed. I didn't touch this piece of code per say.

Reply:
No , point is about the unit tests. Some test should fail right if the conversion fails ? the comment is to add unit tests to detect such changes. 

I see no test failures even when the logic is wrong [ the logic is fixed in latest merge PR yes ]